linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -3072,7 +3099,46 @@ private[optimizer] abstract class OptimizerCore(config: CommonPhaseConfig) {
 
     (op: @switch) match {
       case === | !== =>
-        val positive = (op == ===)
+        // Try to optimize as a primitive JS strict equality
+
+        def canBePrimitiveNum(tpe: RefinedType): Boolean = tpe.base match {
+          case AnyType | ByteType | ShortType | IntType | FloatType | DoubleType =>
+            true
+          case ClassType(cls) =>
+            !tpe.isExact && MaybeHijackedPrimNumberClasses.contains(cls)
```

Comment:
To check my understanding: If the type is exact and a class type of a primitive box, it means the only allowed value is `null` which is not a primitive number. Correct?